### PR TITLE
Feature/regtest address

### PR DESCRIPTION
**Pull request by SuperCipher**
merge to release commit

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -21,7 +21,21 @@ let getBlockchainInfoSync: () => BlockchainInfoResult = sp(RpcClient.getBlockcha
...
+  case 'testnet':
+    network = 'testnet'
+    break
+  case 'regtest':
+    network = 'regtest'
+    break
+  default:
+    network = 'mainnet'
+}
+
```
Comment (litvinof) [inline index.ts:38]: Wouldn't it look cleaner if we change this to look something like

```
const chainsAvailable = [′mainnet′, ′testnet′, ‘regtest‘]
let network = networksAvailable.includes(chain) ? chain : ′mainnet'
```

**Review by litvinof — CHANGES_REQUESTED** [review]
One comment left, other than that - LGTM

File: slptokengraph.ts [hunk 2]
```diff
@@ -572,7 +573,11 @@ export class SlpTokenGraph {
     private getAddressStringFromTxnOutput(txn: bitcore.Transaction, outputIndex: number) {
         let address;
         try {
-            address = Utils.toSlpAddress(bitbox.Address.fromOutputScript(txn.outputs[outputIndex]._scriptBuffer, this._network));
+            if (this._network === 'regtest'){
+              address = Utils.toSlpRegtestAddress(bitbox.Address.fromOutputScript(txn.outputs[outputIndex]._scriptBuffer, 'bchreg'));
+            } else{
```
Comment (nicolaiskye) [inline slptokengraph.ts:578]: Maybe I'm nitpicking, but I think
`} else {`
with an extra space looks more clean.

File: tna.ts [hunk 3]
```diff
@@ -9,7 +9,21 @@ let bitcore = require('bitcore-lib-cash');
 
 export class TNA {
     fromTx(gene: Bitcore.Transaction, options?: any): TNATxn {
-        let net = options.network === 'testnet' ? bitcore.Networks.testnet : bitcore.Networks.livenet;
+        let net : string
```
Comment (nicolaiskye) [inline tna.ts:12]: Same thing here as @litvinof mentioned on the index.ts file.

File: tna.ts [hunk 4]
```diff
@@ -42,17 +56,23 @@ export class TNA {
                         s: input._scriptBuffer,
                     }
                     let address;
-                    try { address = Utils.toSlpAddress(input.script.toAddress(net).toString(bitcore.Address.CashAddrFormat)); } catch(_) { }
+                    try { address = Utils.toSlpAddress(input.script.toAddress(net).toString(bitcore.Address.CashAddrFormat));
+                      if (options.network === 'regtest'){
+                        address = Utils.toSlpRegtestAddress(input.script.toAddress(net).toString(bitcore.Address.CashAddrFormat));
+                      } else{
```
Comment (nicolaiskye) [inline tna.ts:62]: More nitpicking
`} else {`

**Review by nicolaiskye — APPROVED** [review]
LGTM

**Review by litvinof — APPROVED** [review]

**Pull request merged**